src/libcore/iter.rs

```diff
@@ -148,7 +148,7 @@ pub trait Iterator {
         last
     }
 
-    /// Loops through `n` iterations, returning the `n`th element of the
+    /// Loops through `n` iterations, returning the next (`n+1`th) element of the
```

Comment:
What about:

> Loops through `n` iterations, returning the next (`n`th) element of the iterator. The first element returned by the iterator is the "zeroth" one.

Or otherwise, just say it completely differently:

> Returns the element at index `n` by looping through `n` iterations and returning the next element of the iterator.